overlord/ifacestate/handlers.go

```diff
@@ -558,6 +558,14 @@ func (m *InterfaceManager) doAutoConnect(task *state.Task, _ *tomb.Tomb) error {
 	st.Lock()
 	defer st.Unlock()
 
+	// The previous task (link-snap) may have triggered a restart,
+	// if this is the case we can only procceed once the restart
+	// has happend or we may not have all the interfaces of the
+	// new core/base snap.
+	if err := snapstate.WaitRestart(task); err != nil {
+		return err
+	}
```

Comment:
at least for core as bootable base on core we need to check that we came back to the right revision as the old code did, no?

Reply:
Yes, there is also a (integration) test part missing here, I will look into this.